source/s3_auto_ranged_put.c

```diff
@@ -979,6 +991,8 @@ static void s_s3_auto_ranged_put_request_finished(
                 }
 
                 auto_ranged_put->synced_data.list_parts_completed = !has_more_results;
+                auto_ranged_put->synced_data.list_parts_sent =
+                    !has_more_results; /* Unset list_parts_send if there are more to send */
```

Comment:
put the comment on its own line. If your trailing comment forces a line-wrap, it all just gets hard to read

add comments, or something, to explain how this will result in another ListParts request being sent out, using the continuation token.

also the two `bool = !otherbool;` in a row is a bit confusing. Maybe code it like you'd explain it.

```
if (has_more_results) {
    /* explain about how this results in another ListParts request */
    auto_ranged_put->synced_data.list_parts_sent = false;
} else {
    auto_ranged_put->synced_data.list_parts_completed = true;
}